manifoldco/ui
# MVP for killing the loops

## Reason for change



Our network calls can sometimes kick off before we have an auth token, so we currently employ a crude polling mechanism to account for this. It checks for a token every 2 seconds until the specified wait time is met or exceeded.

This change employs an event listener instead of a loop, so that the network call can commence as soon as the token is actually received. It will still timeout after the wait time is met, but without having to guess at a reasonable polling interval.

## Testing


This only behaves differently when there is no token provided to `manifold-auth-token`. You can test this in Render dashboard:

1. Go to the Addons page
1. Remove the `manifold_api_token` from local storage
1. Refresh the page
1. You should eventually see your addons

The difference from the current behavior will strictly be a speed increase: on average, we should see our addons sooner with this change than we do without it.

## Checklist




[docs]: https://ui.sandbox.manifold.co
